example_allocators/compartment_alloc/compartment_alloc.c

```diff
@@ -0,0 +1,91 @@
+#include <cheriintrin.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+#include "compartment_alloc.h"
+#include "utils.h"
+
+heap_compartment compartments[maxCompartments];
+int numCompartments = 0;
+
+/*
+ * Initialise a compartment with `size_in_bytes` space, where all future
+ * allocations will be constrained by `dc` as if it was the DDC when allocation
+ * occurred.
+ */
+void
+*__capability init_compartment(int size_in_bytes, void *__capability dc)
+{
+    // We impose a limit on our # of compartments
```

Comment:
Why do we impose a limit?

Reply:
Just because we've allocated a limited number in the `compartments` array, no other reason. Maybe I should remove the comment…?

Reply:
Yeah, we might as well remove the comment. For an example we don't need to justify *why* we're being restrictive: it's an example :)

Reply:
Yeah, probably best to remove it here for the example, or at least give some justification if you want to keep it.

Reply:
Removed in [c37081b](https://github.com/capablevms/cheri-examples/pull/88/commits/c37081bbcec4a81ff86cb083260dbddca70dc80d), thanks both!